dashboard/test/ui/test_status.js

```diff
@@ -176,20 +180,7 @@ function testFromS3Key(key) {
     return undefined;
   }
 
-  // Incrementally replace underscores with slashes and try again, in case
-  // this was a test from a subdirectory.
-  // Example:
-  //   Feature: features/applab/tooltips.feature
-  //   Key    : {browser}_applab_tooltips_output.html
-  // If we ever have an ambiguous case (someone creates
-  // features/applab_tooltips.feature) then this will be a problem - but it is
-  // unlikely.
-  var test = tests[browser][feature];
-  while (!test && feature.indexOf('_') >= 0) {
-    feature = feature.replace(/_/, '/'); // Replaces first underscore only.
-    test = tests[browser][feature];
-  }
```

Comment:
We ran into a similar problem a long time ago: Our test runner uses underscores in output filenames, so feature `applab/tooltips.feature`'s results are stored in `<browser>_applab_tooltips_output.html`.  When the status page is generated from feature names, but the status we get back is in output file names; so, we do a reverse mapping with `testFromS3Key` to apply the status to the appropriate test row.

This solution that [I wrote two years ago](https://github.com/code-dot-org/code-dot-org/commit/e1025abaa65c04ed603d3f9d0f6a03f1b146e442) :older_man: iteratively replaces underscores in the output name with slashes, attempting to reconstruct the folder structure.  But in our case it doesn't work:
```
// Real feature filename
public_key_cryptography/continue_button.feature
// S3 key
<browser>_public_key_cryptography_continue_button_output.html
// Attempted reconstructions:
public_key_cryptography_continue_button.feature
public/key_cryptography_continue_button.feature
public/key/cryptography_continue_button.feature
public/key/cryptography/continue_button.feature
public/key/cryptography/continue/button.feature
```

In essence, the algorithm assumes there will never be an underscore in any directory names.

My solution is to invert the approach.  When we set up our `Test` objects we strip all slashes from the key, so we're actually using `public_key_cryptography_continue_button.feature` for the lookup - then we do the same here.

There's a chance of ambiguity with this approach (`applab/tooltips.feature` vs `applab_tooltips.feature`) but that chance existed with the previous solution too (see comment) so I don't feel bad about it.

Reply:
I think this is fine and the chance of ambiguity is low, but if we wanted to be more cautious what about using a different symbol for directories, even something like two underscores? Then `applab/tooltips.feature` would map to `applab__tooltips.feature`, and would not conflict with `applab_tooltips.feature`.